### DISCUSSION TITLE
vercel.json rewrites do not work for svelte

### DISCUSSION THREAD

**Discussion by magicwrites** [vercel/community#1045] [Help] [answered]
Hello everyone!

On [my svelte based website](https://mazur.site/) I need to implement rewrite or redirect rules exactly as described in this [article](https://vercel.com/guides/can-i-redirect-from-a-subdomain-to-a-subpath).

I ended up with [this vercel.json configuration](https://github.com/magicwrites/site-mazur/blob/master/vercel.json) for my particular case with 4 subdomains soon to be turned redirects - [maciej.mazur.site](https://maciej.mazur.site/) which should lead to [/maciej](https://www.mazur.site/maciej) is a fair example.

I have read the article a few times, experimented with wildcards and domain settings - unfortunately nothing works.
My best guess right now is that vercel.json does not work at all.

Most of the related discussions that I could find point towards next.js based solutions - unfortunatelly I do not use that.

My domain settings essentially involves the subdomains with `No Redirect` selected.
~`helena.mazur.site` subdomain case is missed on purpose to make sure domain settings have nothing to do with it.~ not valid anymore

![image](https://user-images.githubusercontent.com/1070654/201491194-251bc3f8-c63d-498a-ab44-6d99d2db3fbf.png)

Can I kindly ask for a second pair of eyes on this? The [source code](https://github.com/magicwrites/site-mazur/blob/master/vercel.json) is publicly available.
Is there any log related to `vercel.json` execution that I could use for debugging?

**Comment by magicwrites** [comment]
I have now [bypassed the issue by providing the svelte redirection](https://github.com/magicwrites/site-mazur/blob/master/src/routes/%2Blayout.server.ts) so expect my problem to be gone - but I will keep the vercel.json code for a while so that somebody might point out the error in my vercel configuration.

**Comment by amyegan** [comment, marked as answer]
Hi @magicwrites! I think the `vercel.json` you shared will result in a sort of infinite redirect. Your [SvelteKit redirects](https://kit.svelte.dev/docs/load#redirects) solution seems like a better option. Otherwise, you would need to add a sort of regex bit to the `source` value or rewrite/redirect to something like `https://mazur.site/:path*` to remove the subdomain prefix after a match occurs.
